Infrastructure: Add missing test for 6 examples

spectranaut opened the pull request:
https://github.com/w3c/aria-practices/issues/1378

This PR adds tests for: 
- [the button idl example](https://w3c.github.io/aria-practices/examples/button/button_idl.html)
- Fixes some references in the combobox grid text file
- Adds some missing tests for the rest of the combobox tests
- Adds the missing keyboard interaction tests to the spin button date picker

nschonni commented:
@spectranaut it would be good to rebase this one to see how the regression test comment job handles this one

spectranaut commented:
@jongund, any chance you could code review this PR? It would be nice to get the missing tests down to zero :)

nschonni commented:
@spectranaut looks like the prettier stuff caused some conflicts

jongund commented:
@spectranaut 

Valarie, did you add the missing tests or just reformat the existing test files to use "prettier"?

spectranaut commented:
> Valarie, did you add the missing tests or just reformat the existing test files to use "prettier"?

 @jongund I wrote some tests! I'm not sure why I have a failure on the PR for the js linter though.

nschonni commented:
@spectranaut there are some in-line comments on the file tab around `'use strict' is unnecessary inside of modules` for the linting failures

spectranaut commented:
@nschonni yeah I'm a bit baffled by them, there aren't errors for the use of "no strict" in other files, I can't replicate locally...

nschonni commented:
They got removed with https://github.com/w3c/aria-practices/pull/1566
I'm guessing you might need to rebase and/or maybe re-run `npm ci` locally

jongund commented:
> 
> 
> @nschonni yeah I'm a bit baffled by them, there aren't errors for the use of "no strict" in other files, I can't replicate locally...

I have had this problem with 'strict' in the past too, I don't remember how I fixed it.

nschonni commented:
OK, tested it out locally:
- checkout locally and run `npm ci`. `npm run fix` had no changes or errors
- checkout locally, rebase with master, and run `npm ci`. `npm run fix` updated the 2 flagged files

I think the checkout for CI is using the merge commit onto main, so that's why it's showing the errors there and not with just the local branch merge

spectranaut commented:
@nschonni I think we did a little duplicate work, but thanks for look into it :) I rebased and fixed and force pushed and the linter now passes. Looking into the regression test failure now.

spectranaut commented:
Ok I'm a bit confused by the PR and the failures I'm seeing. It's a time out waiting for a link when testing "key enter" on the link apg example -- which is code this PR does not touch. We have seen this error before and guessed it had to do with the travis check environment's resources in some case, or sending to many requests to wikipedia in another case. I tried testing just link over on the bocoup branch (https://github.com/bocoup/aria-practices/pull/43) with some console logs to try to get a better sense of what is happening, and it failed on time and passed the next.

Anyway, on the most recent run off all the checks it passes. I'm going to open a backlog issue to track these link related failures in case they come up again.

nschonni commented:
Maybe another 429 response code like the wikipedia ones for the tab tests before?

rwaldron reviewed (approved):
I was asked (by @spectranaut) to provide a peer review to help move this forward. I've read through all of the changes here and nothing stands out. When I tested locally, I experienced one test failure that was distinct from failures I experienced when running the test from the master branch, and which is only reproducible on my local machine. I don't consider this a blocking issue. 

I've followed the [review guidelines](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#test_review) and have determined that this work satisfies all of the criteria laid out in that document. 

(I'd like to acknowledge that I do not have the same level of familiarity with the project as the regular contributors)

rwaldron commented on test/tests/button_button-idl.js:
```diff
@@ -0,0 +1,168 @@
+const ex = {
+  printSelector: '#action',
+  toggleSelector: '#toggle',
+  svg: '#example svg',
```
For consistency, I believe this should be named `svgSelector`?

jongund commented:
@rwaldron 
Thank you for the feedback, but right now I am focused on finishing up some other ARIA example test cases, but when I get to toolbar I will review your comments.

spectranaut commented:
Thanks @rwaldron ! I adopted your feedback, including simplifying one test that remove the failure you got when you run it locally.

spectranaut commented:
The failing test is due to the flaky link tests currently being fixed in #1600
